pydantic/_internal/_generate_schema.py

```diff
@@ -816,14 +833,18 @@ def _generate_schema_from_property(self, obj: Any, source: Any) -> core_schema.C
 
         schema: CoreSchema
 
-        if (get_schema := getattr(obj, '__get_pydantic_core_schema__', None)) is not None:
+        if (
+            (get_schema := getattr(obj, '__get_pydantic_core_schema__', None)) is not None
+            and self._rebuild_nested_model_schemas == False
```

Comment:
This is only necessary so that the logic that walks the built schema to find the discriminants of unions can recurse into models. I think ideally we would handle this some other way. I imagine this is responsible for the `test_nested_model_schema_generation` regression